ingestion/functions/parsing/japan/japan.py

```diff
@@ -0,0 +1,87 @@
+import json
+import os
+import sys
+from datetime import date, datetime
+
+# Layer code, like parsing_lib, is added to the path by AWS.
+# To test locally (e.g. via pytest), we have to modify sys.path.
+# pylint: disable=import-error
+if ('lambda' not in sys.argv[0]):
+    sys.path.append(
+        os.path.join(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+            'common/python'))
+import parsing_lib
+
+def convert_gender(raw_gender):
+    if "gender" in raw_gender:
+        if raw_gender["gender"] == "M":
+            return "Male"
+        if raw_gender["gender"] == "F":
+            return "Female"
+
+def convert_age(raw_age):
+    age_range = {}
+    if raw_age["ageBracket"] != -1:
+        age_range["start"] = raw_age["ageBracket"]
+        age_range["end"] = raw_age["ageBracket"] + 9
+        return age_range
+    else:
+        return None
+
+def convert_location(raw_entry):
+    query_terms = [
+        term for term in [
+            raw_entry.get("detectedCityTown", ""),
+            raw_entry.get("detectedPrefecture", ""),
+            "Japan"]
+        if term and term != "Unspecified"]
+    return {"query":  ", ".join(query_terms)}
+
+def detect_notes(raw_notes):
+    if "notes" in raw_notes:
+        return raw_notes["notes"]
+    else:
+        return None
+
+def parse_cases(raw_data_file, source_id, source_url):
+    """
+    Parses G.h-format case data from raw API data.
+    """
+    with open(raw_data_file, "r") as f:
+        cases = json.load(f)
+        return [
+            {
+                "caseReference": {
+                    "sourceId": source_id,
+                    "sourceEntryId": entry["patientId"],
+                    "sourceUrl": source_url
+                },
+                "revisionMetadata": {
+                    "revisionNumber": 0,
+                    "creationMetadata": {
+                        "curator": "auto",
+                        "date": date.today().strftime("%m/%d/%Y")
+                    }
+                },
```

Comment:
A separate PR, but we should move this into the library. I can't recall -- is this part used, still, or does the service account identity get inferred/used based on it supplying the bearer token?

Reply:
oh I missed that, it's not needed at all, removed.